darwin/client.py

```diff
@@ -386,19 +182,152 @@ def create_dataset(self, name: str, team: Optional[str] = None) -> RemoteDataset
             client=self,
         )
 
-    def fetch_remote_classes(self, team: Optional[str] = None) -> Optional[Dict[str, Any]]:
-        """Fetches all remote classes on the remote dataset"""
+    def archive_remote_dataset(self, dataset_id: int, team_slug: str) -> None:
+        """
+        Archive (soft delete) a remote dataset.
+
+        Parameters
+        ----------
+        dataset_id: int
+            Id of the dataset to archive.
+        team_slug: str
+            Team slug of the dataset.
+        """
+        self._put(f"datasets/{dataset_id}/archive", payload={}, team=team_slug)
```

Comment:
We're probably not using the response when we call `archive_remote_dataset`, but I would actually return the response for API consistency. What do you think?

Reply:
I would go even further and actually use a `DatasetIdentifier` iin every place we are using a slug or an `id`. However, I think I made enough breaking changes as is for 1 PR :D

Reply:
Ticket created for this topic:
https://linear.app/v7labs/issue/V7-2547/harmonization-part-2